pages/resources/quizzes/FirstLang/FirstProgrammingLanguageQuiz.tsx

```diff
@@ -0,0 +1,116 @@
+import { useEffect, useState } from "react";
+import LangResults from "../../../../components/resources/quizzes/langQuiz/LangResults";
+import BluePrint from "../../../../components/resources/quizzes/shared/BluePrint";
+import SkillSelections, {
+  QuizOptionViewState,
+  QuizViewState,
+} from "../../../../components/resources/quizzes/shared/SkillSelections";
+import StartQuiz from "../../../../components/resources/quizzes/shared/StartQuiz";
+import { quizData } from "../../../api/studentPortal/quizzes/firstProgrammingLanguage";
+
+export enum Stage {
+  START,
+  QUESTIONS,
+  BLUEPRINT,
+  RESULTS,
+}
+
+const FirstProgrammingLanguageQuiz = () => {
+  const [stage, setStage] = useState(Stage.START);
+  const [quizViewState, setQuizViewState] = useState<QuizViewState>();
+
+  useEffect(() => {
+    const quizViewState = {
+      title: quizData.title,
+      body: quizData.body,
+      questions: [],
```

Comment:
instead of initializing this prop with an empty array and pushing questions into this array you could use the map function to clean this up and remove the for loop.

ie. 

```
questions: quizData.questions.map(question => {
  return {
    title: question.title,
    body: question.body,
    options: question.options,
  }
}
```

great job figuring out the for loop approach, but you need to start thinking about data as functional instead of imperative. Imperative programming is very junior. You think of code as being executed one line at a time.

To get yourself to the next level, you need to think of arrays as a stream of data that can be filtered or mapped. This is a more functional way of thinking about code.

Reply:
@LuckyHariharan you should have noticed this as well.

Reply:
awesome! I didn't think of this!